api/envoy/extensions/filters/http/gcp_authn/v3/gcp_authn.proto

```diff
@@ -24,17 +26,39 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 message GcpAuthnFilterConfig {
   // The HTTP URI to fetch tokens from GCE Metadata Server(https://cloud.google.com/compute/docs/metadata/overview).
   // The URL format is "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/identity?audience=[AUDIENCE]"
-  config.core.v3.HttpUri http_uri = 1 [(validate.rules).message = {required: true}];
+  //
+  // This field is deprecated because it does not match the API surface provided by the google auth libraries.
+  // Control planes should not attempt to override the metadata server URI.
+  // The cluster and timeout can be configured using the `cluster` and `timeout` fields instead.
+  // For backward compatibility, the cluster and timeout configured in this field will be used
+  // if the new `cluster` and `timeout` fields are not set.
+  config.core.v3.HttpUri http_uri = 1 [deprecated = true, (envoy.annotations.deprecated_at_minor_version) = "3.0"];
 
   // Retry policy for fetching tokens. This field is optional.
+  // Not supported by all data planes.
   config.core.v3.RetryPolicy retry_policy = 2;
 
   // Token cache configuration. This field is optional.
   TokenCacheConfig cache_config = 3;
 
   // Request header location to extract the token. By default (i.e. if this field is not specified), the token
   // is extracted to the Authorization HTTP header, in the format "Authorization: Bearer <token>".
-  TokenHeader token_header = 4;
+  //
+  // This field is deprecated because it does not match the API surface provided by the google auth libraries.
+  // Control planes should not attempt to override the token header.
+  TokenHeader token_header = 4 [deprecated = true, (envoy.annotations.deprecated_at_minor_version) = "3.0"];
```

Comment:
Looks like there could be OSS users using this field like here https://github.com/envoyproxy/envoy/issues/28894#issuecomment-1675159504. If gRPC doesn't need this field, can gRPC just leave this unset?

I will defer this to API reviewer @htuch

Reply:
Yeah, not sure why we are deprecating fields based purely on what a given library supports.

Reply:
This is less about what individual data planes may implement and more about making this API consistent with that of the google auth libraries.  Most gRPC implementations will wind up delegating to those existing auth libraries rather than reimplementing this functionality themselves, and I would expect most other data planes to do the same thing.  (If it weren't for the fact that C++ is one of the few languages for which there is no google auth library, I would have expected Envoy to have done the same thing, in which case this issue would never have come up.)

That having been said, I feel less strongly about this for `token_header` than I do for `http_uri`.  I'll go ahead and un-deprecate this field but add a comment that not all data planes support it, and we will not support it in gRPC.